src/nbl/CMakeLists.txt

```diff
@@ -394,7 +394,11 @@ macro(nbl_target_link_volk _trgt)
 	target_link_libraries(${_trgt} PRIVATE volk)
 endmacro()
 macro(nbl_target_link_EGL _trgt)
-	target_link_libraries(${_trgt} PUBLIC egl)
+	if (${_NBL_USE_SYSTEM_EGL}) # in case of _NBL_USE_SYSTEM_EGL, we need just egl.h for EGL function types and constants
```

Comment:
not a compile time switch

Reply:
eh i got your code snippet on discord a little wrong, easy to change
can I assume one of WGL or GLX is always available then? I mean if not, then I have to do some simple trick to make sure egl functions are always defined (at least just empty return null) to be able to compile taking ptrs to them

Reply:
WGL is always available (unless something is vastly different on Windows Server)

GLX is not always available:
- Wayland session = no X11
- X11 session = no Wayland
- terminal session = no X11 or Wayland

> I have to do some simple trick to make sure egl functions are always defined (at least just empty return null) to be able to compile taking ptrs to them

EGL has a header that comes from Khronos, the declarations that is.

The function pointers could just be null?